packages/fxa-settings/src/lib/gql.ts

```diff
@@ -28,9 +28,19 @@ export const errorHandler: ErrorHandler = ({ graphQLErrors, networkError }) => {
       if (error.message === 'Invalid token') {
         reauth = true;
       } else if (error.message === 'Must verify') {
-        return window.location.replace(
-          `/signin_token_code?action=email&service=sync`
-        );
+        // Any place we reference `account`, like `account.sendVerificationCode()` in
+        // ConfirmSignupCode, (over)fetches account data. When we finish refactoring
+        // ConfirmSignupCode to the container component pattern, this will no longer
+        // be a problem, but for now, we do not want to redirect on that page when
+        // there's an authentication problem; we just want to redirect to /signin
+        // if the account is not in localStorage, which is handled in ConfirmSignupCode
+        // container. So, for now, we don't care about that page here. We'll still
+        // likely want to modify this for cases where the service is NOT sync.
+        if (!window?.location.pathname.includes('confirm_signup_code')) {
```

Comment:
Nit: Do we need the `?.` on window? In general this seems pretty inconsistent in this file. I see another place where it's `window?.loaction?.pathname` and another place where it's just `window.location.`. No concrete advice here... Just wanted to point this out.

Reply:
I'll clean this up for consistency with `?` for both window and location where they're first checked. The instance of `window.location` is fine as we only reach that line if both window and location are defined.